### ISSUE TITLE
Mode absent sur thermostat th1123wf dashboard

### ISSUE THREAD

**Issue by simtreti** [claudegel/sinope-130#111]
Bonjour Claude,

Il semble avoir un petit bug d'affichage du statut des thermostats, pour un thermostat th1123wf qui est en mode absent dans la carte thermostat sur le dashboard. Il affiche qu'il est en mode Heat alors qu'il devrait afficher Away (Garde Robe)
Dans la capture d'écran ci-dessous tous mes autre thermostst des th1120rf sont en mode absent et on le voit bien. 
![IMG_3060](https://user-images.githubusercontent.com/79706737/200417173-34c28e1a-f002-4606-a207-47bc06d5b822.PNG)
Est-ce possible de corriger ce comportement ?

**Comment by claudegel** [comment]
Bonjour,
Je sais que le mode away est different entre les thermostats wifi et les zigbee. Il y a un bout de temps que j'ai joué avec ça. Il faut investiguer mais je n'ai pas de TH1123WF et en fait aucun appareil wifi sur mon HA. 
Donc il va me falloir un peu d'aide. Premièrement je voudrait voir ce qu'il y a sur Neviweb. Pour les zigbee les modes sont seulement chauffage et arret. 
![mode zigbee](https://user-images.githubusercontent.com/4268938/200462965-7aaec877-6689-4452-a18b-20e22cce2cd3.jpg)
Ensuite il me faut les log de HA pour votre TH1123WF.
recherchez la ligne qui contient ceci
[custom_components.neviweb130.climate] Updating neviweb130 climate nom_du_thermostat_wifi

**Comment by claudegel** [comment]
Quel est la valeur de l'attribut occupancy pour votre TH1123WF dans HA quand vous êtes en away

**Comment by simtreti** [comment]
Bonjour Claude,

Voici les modes dans neviweb
<img width="448" alt="Capture d’écran, le 2022-11-07 à 21 53 01" src="https://user-images.githubusercontent.com/79706737/200466056-810df09b-d1b6-4f23-a1ca-fb55a9b7865b.png">
L'attribut occupancy est comme ceci: occupancy: away

voici le log. 

2022-11-07 22:12:45.658 DEBUG (SyncWorker_6) [custom_components.neviweb130.climate] Updating neviweb130 climate Garde Robe (0.715 sec): {'roomSetpoint': 18, 'outputPercentDisplay': {'percent': 0, 'sourceType': 'heating'}, 'roomTemperature': {'value': 21.73, 'error': None}, 'roomSetpointMin': 5, 'roomSetpointMax': 30, 'roomTemperatureDisplay': {'value': 22, 'status': 'on'}, 'timeFormat': '24h', 'temperatureFormat': 'celsius', 'drStatus': {'drActive': 'off', 'optOut': 'off', 'setpoint': 'off', 'powerAbsolute': 'off', 'powerRelative': 'off'}, 'drSetpoint': {'status': 'off', 'value': None}, 'loadWattOutput1': 1834, 'loadWatt': 1834, 'wifiRssi': -58, 'keyboardLock': 'unlock', 'config2ndDisplay': 'setpoint', 'setpointMode': 'auto', 'occupancyMode': 'home', 'backlightAutoDim': 'alwaysOn', 'earlyStartCfg': 'off', 'roomSetpointAway': 18}
2022-11-07 22:12:49.748 DEBUG (SyncWorker_0) [custom_components.neviweb130] Data = {'occupancyMode': 'away'}
2022-11-07 22:12:49.748 DEBUG (SyncWorker_0) [custom_components.neviweb130] Request response = 200
2022-11-07 22:12:49.748 DEBUG (SyncWorker_0) [custom_components.neviweb130] Json Data received= {'occupancyMode': 'away'}
2022-11-07 22:12:49.748 DEBUG (SyncWorker_0) [custom_components.neviweb130] Content = b'{"occupancyMode":"away"}'
2022-11-07 22:12:49.748 DEBUG (SyncWorker_0) [custom_components.neviweb130] Text = {"occupancyMode":"away"}
2022-11-07 22:12:49.833 DEBUG (SyncWorker_8) [custom_components.neviweb130.climate] Updating neviweb130 climate Garde Robe (0.084 sec): {'roomSetpoint': 18, 'outputPercentDisplay': {'percent': 0, 'sourceType': 'heating'}, 'roomTemperature': {'value': 21.73, 'error': None}, 'roomSetpointMin': 5, 'roomSetpointMax': 30, 'roomTemperatureDisplay': {'value': 22, 'status': 'on'}, 'timeFormat': '24h', 'temperatureFormat': 'celsius', 'drStatus': {'drActive': 'off', 'optOut': 'off', 'setpoint': 'off', 'powerAbsolute': 'off', 'powerRelative': 'off'}, 'drSetpoint': {'status': 'off', 'value': None}, 'loadWattOutput1': 1834, 'loadWatt': 1834, 'wifiRssi': -58, 'keyboardLock': 'unlock', 'config2ndDisplay': 'setpoint', 'setpointMode': 'auto', 'occupancyMode': 'away', 'backlightAutoDim': 'alwaysOn', 'earlyStartCfg': 'off', 'roomSetpointAway': 18}

**Comment by claudegel** [comment]
Donc vous pouvez mettre le thermostat en mode away mais ceci n'apparait pas sur l'affichage du thermostat dans HA. Le preset_mode ne s'affiche pas.
Je vais faire quelques test

**Comment by simtreti** [comment]
Exact, c'est seulement l'affichage qui est erroné dans le dashboard.

**Comment by claudegel** [comment]
Désolé j'ai un petit rush à terminer d'ici jeudi matin. Mais ça ne devrait pas être difficile à corriger. Dès que j'ai une patch je vais vous aviser pour que vous la testiez

**Comment by claudegel** [comment]
Bon voici je pense que j'ai corrigé le bug. J'ai créé une branche «preset»
https://github.com/claudegel/sinope-130/tree/preset
Copiez le fichier climate.py dans config/custom_components/neviweb130 et redémarrez HA
Ca devrait afficher le bon preset dans la carte de vos thermostats

**Comment by simtreti** [comment]
Bonsoir

maintenant le thermostat ne fonctionne plus au restart de HA

2022-11-10 21:36:01.324 ERROR (MainThread) [homeassistant.loader] Unexpected exception importing platform custom_components.neviweb130.climate
2022-11-10 21:36:01.325 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform neviweb130.climate: Platform not found (Exception importing custom_components.neviweb130.climate).

**Comment by claudegel** [comment]
Vous avez copié le fichier climate.py dans votre repertoire neviweb130 ?

**Comment by simtreti** [comment]
Oui, oublié mon dernier post tout fonctionne. J'avais trop de session en cours tous fonctionne bien maintenant ! le preset Away s'affiche !

**Comment by claudegel** [comment]
Parfait je vais faire un release pour que tout le monde en profite

Merci d'avoir signalé ce bug et pour votre aide à tester car je n'ai pas de thermostat wifi sur mon réseau.

**Comment by simtreti** [comment]
Merci à vous !
